db/sql/prototype.sql

```diff
@@ -0,0 +1,491 @@
+*/
+BEGIN
+  RETURN __msar.exec_ddl(
+    'SELECT '
+      || 'setval('
+      || 'pg_get_serial_sequence(''%1$s'', ''%2$s''), coalesce(max(%2$s) + 1, 1), false'
+      || ') '
+      || 'FROM %1$s',
+    table_name, column_
+  );
+END;
+$$ LANGUAGE plpgsql RETURNS NULL ON NULL INPUT;
+
+
+CREATE OR REPLACE FUNCTION
+msar.update_pk_sequence_to_latest(table_id oid, col_attnum integer) RETURNS text AS $$/*
+Update the primary key sequence to the maximum of the primary key column, plus one.
+
+Args:
+  table_id: The OID of the table whose primary key sequence we'll update.
+  col_attnum: The attnum of the primary key column.
+*/
+DECLARE table_name text;
+DECLARE colname text;
+BEGIN
+  table_name :=  __msar.get_table_name(table_id);
+  colname := msar.get_column_name(table_id, col_attnum);
+  RETURN __msar.update_pk_sequence_to_latest(qualified_table_name, colname);
+END;
+$$ LANGUAGE plpgsql RETURNS NULL ON NULL INPUT;
+
+
+CREATE OR REPLACE FUNCTION
+msar.update_pk_sequence_to_latest(schema_ text, table_name text, column_ text) RETURNS text AS $$/*
+Update the primary key sequence to the maximum of the primary key column, plus one.
+
+Args:
+  table_id: The OID of the table whose primary key sequence we'll update.
+  col_attnum: The attnum of the primary key column.
+*/
+DECLARE qualified_table_name text;
+BEGIN
+  qualified_table_name := msar.get_fq_table_name(schema_, table_name);
+  RETURN __msar.update_pk_sequence_to_latest(qualified_table_name, quote_ident(column_));
+END;
+$$ LANGUAGE plpgsql RETURNS NULL ON NULL INPUT;
+
+
+----------------------------------------------------------------------------------------------------
+----------------------------------------------------------------------------------------------------
+-- MATHESAR VIEW FUNCTIONS
+--
+-- Functions and triggers in this section are used to create a view that tracks any created or
+-- modified table, and also provide a convenient way to set up such a view for an already-existing
+-- table or drop such a view.
+----------------------------------------------------------------------------------------------------
+----------------------------------------------------------------------------------------------------
+
+CREATE OR REPLACE FUNCTION
+msar.create_mathesar_view(table_id oid) RETURNS text AS $$/*
+Create a view of named mv<table_id> tracking the table with OID <table_id>.
+
+Args:
+  table_id: This is the OID of the table we want to track.
+
+*/
+DECLARE viewname text;
+DECLARE viewcols text;
+BEGIN
+  viewname := msar.get_msar_view_name(table_id);
+  SELECT string_agg(format('%s AS c%s', quote_ident(attname), attnum), ', ')
+    FROM pg_attribute
+    WHERE attrelid=table_id AND attnum>0 AND NOT attisdropped
+  INTO viewcols;
+  RETURN __msar.exec_ddl(
+    'CREATE OR REPLACE VIEW %s AS SELECT %s FROM %s',
+    viewname, viewcols, __msar.get_table_name(table_id)
+  );
+END;
+$$ LANGUAGE plpgsql RETURNS NULL ON NULL INPUT;
```

Comment:
Do I understand this correctly? A Mathesar view is a reflection of a user table. A Mathesar view is algorithmically named according to the user table's oid. Its column values are the reflected user table's column values, and its columns' names are the corresponding reflected table columns' attnums. Would be great to put this in a docstring somewhere around here.

Reply:
You understand correctly. I'll figure out some reasonable way to document the purpose of these views, and how they work.